cmd/revision.go

```diff
@@ -0,0 +1,110 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/databus23/helm-diff/diff"
+	"github.com/databus23/helm-diff/manifest"
+	"github.com/mgutz/ansi"
+	"github.com/spf13/cobra"
+	"k8s.io/helm/pkg/helm"
+)
+
+const revisionCmdLongUsage = `
+This command compares the manifests details of a named release.
+
+It can be used to compare the manifests of 
+ 
+ - lastest REVISION with specified REVISION
+	$ helm diff revision [flags] [RELEASE] [REVISION]
+
+ - REVISION1 with REVISION2
+	$ helm diff revision [flags] [RELEASE] [REVISION1] [REVISION2]
+`
+
+func revisionCmd() *cobra.Command {
+	diff := diffCmd{}
+	revisionCmd := &cobra.Command{
+		Use:   "revision [flags] [RELEASE] REVISION1 REVISION2",
+		Short: "Shows diff between revision's manifests",
+		Long:  revisionCmdLongUsage,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if v, _ := cmd.Flags().GetBool("version"); v {
+				fmt.Println(Version)
+				return nil
+			}
+
+			switch {
+			case len(args) < 2:
+				return errors.New("Too few arguments to Command \"revision\".\nMinimum 2 arguments required: release name, revision")
+			case len(args) > 3:
+				return errors.New("Too many arguments to Command \"revision\".\nMaximum 3 arguments allowed: release name, revision1, revision2")
+			}
+
+			if q, _ := cmd.Flags().GetBool("suppress-secrets"); q {
+				diff.suppressedKinds = append(diff.suppressedKinds, "Secret")
+			}
+
+			if nc, _ := cmd.Flags().GetBool("no-color"); nc {
```

Comment:
This can be a global flag for sure, that is shared by all sub commands. And have the effect for done in a `Pre` function once for all commands.

Reply:
we could use PersistentPreRunE,  
when we add those flag checks PersistentPreRunE  in the root command, all its children commands (upgrade,revision,rollback) will inherit and execute.

Reply:
I think for this flag its a good fit as it just flips a package level var

Reply:
Works fine for me.., hope this is ok

```go
func New() *cobra.Command {

	chartCommand := newChartCommand()

	cmd := &cobra.Command{
		Use:   "diff",
		Short: "Show manifest differences",
		// parse the flags and check for actions like suppress-secrets, no-colors
		PersistentPreRun: func(cmd *cobra.Command, args []string) {
			if nc, _ := cmd.Flags().GetBool("no-color"); nc {
				ansi.DisableColors(true)
			}
		},
		RunE: func(cmd *cobra.Command, args []string) error {
			cmd.Println(`Command "helm diff" is deprecated, use "helm diff upgrade" instead`)
			return chartCommand.RunE(cmd, args)
		},
	}
```